rotkehlchen/chain/optimism/modules/walletconnect/balances.py

```diff
@@ -0,0 +1,60 @@
+import logging
+from collections import defaultdict
+from typing import TYPE_CHECKING
+
+from rotkehlchen.accounting.structures.balance import Balance, BalanceSheet
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.chain.ethereum.interfaces.balances import BalancesSheetType, ProtocolWithBalance
+from rotkehlchen.chain.optimism.modules.walletconnect.constants import (
+    CPT_WALLETCONNECT,
+    WCT_TOKEN_ID,
+)
+from rotkehlchen.constants import ZERO
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.optimism.decoding.decoder import OptimismTransactionDecoder
+    from rotkehlchen.chain.optimism.node_inquirer import OptimismInquirer
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class WalletconnectBalances(ProtocolWithBalance):
+    def __init__(
+            self,
+            evm_inquirer: 'OptimismInquirer',
+            tx_decoder: 'OptimismTransactionDecoder',
+    ):
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            tx_decoder=tx_decoder,
+            counterparty=CPT_WALLETCONNECT,
+            deposit_event_types={  # TODO: we use deposit/remove to calculate balance so perhaps deposit_event_types should be renamed?  # noqa: E501
+                (HistoryEventType.STAKING, HistoryEventSubType.DEPOSIT_ASSET),
+                (HistoryEventType.STAKING, HistoryEventSubType.REMOVE_ASSET),
+            },
+        )
+
+    def query_balances(self) -> 'BalancesSheetType':
+        """Query balances of staked WalletConnect"""
+        balances: BalancesSheetType = defaultdict(BalanceSheet)
+        wct_token = Asset(WCT_TOKEN_ID)
+        wct_price = Inquirer.find_usd_price(Asset(WCT_TOKEN_ID))
```

Comment:
Wouldn't it make sense to query the price after you check that there are accounts with balances?

Reply:
yes it would